controllers/metal3machine_controller.go

```diff
@@ -151,6 +145,26 @@ func (r *Metal3MachineReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, re
 		return ctrl.Result{}, errors.Wrapf(err, "failed to create helper for managing the machineMgr")
 	}
 
+	// Return early if the BMMachine or Cluster is paused.
+	if util.IsPaused(cluster, capm3Machine) {
+		machineLog.Info("reconciliation is paused for this object")
+		return ctrl.Result{Requeue: true, RequeueAfter: requeueAfter}, nil
+	}
+
+	// set pause annotation on associated bmh (if any)
+	if cluster.Spec.Paused {
+		err := machineMgr.SetPauseAnnotation(ctx)
+		if err != nil {
+			machineLog.Info("failed to set pause annotation on associated bmh")
+			return ctrl.Result{}, nil
+		}
+	} else {
+		err := machineMgr.CheckPauseAnnotation(ctx)
+		if err != nil {
+			machineLog.Info("failed to check pause annotation on associated bmh")
+			return ctrl.Result{}, nil
+		}
+	}
```

Comment:
If your cluster is paused, this will never be executed, due to the return. you need to change the order with L 149

Reply:
True, good catch, changed the code.